modules/drivers/mongo/test/metabase/driver/mongo/query_processor_test.clj

```diff
@@ -153,6 +153,101 @@
+                 "name" "$name",
+                 "price" "$price"}}
+               {"$limit" 5}],
+              :collection "venues",
+              :mbql? true}
+             (qp/query->native
+               (mt/mbql-query venues
+                              {:filters     [[:expression "bob"] [:expression "cobb"]]
+                               :expressions {:bob   [:abs $latitude]
+                                             :cobb [:upper $name]}
+                               :limit       5})))))
+    (testing "Should be able to deal with 2-arity functions"
+      (is (= {:projections '("_id" "name" "category_id" "latitude" "longitude" "price" "bob"),
+              :query
+              [{"$project"
+                {"_id" "$_id",
+                 "name" "$name",
+                 "category_id" "$category_id",
+                 "latitude" "$latitude",
+                 "longitude" "$longitude",
+                 "price" "$price",
+                 "bob" {"$add" ["$price" 300]}}}
+               {"$limit" 5}],
+              :collection "venues",
+              :mbql? true}
+             (qp/query->native
+               (mt/mbql-query venues
+                              {:filters     [[:expression "bob"] [:expression "cobb"]]
+                               :expressions {:bob   [:+ $price 300]}
+                               :limit       5})))))
+    (testing "Should be able to deal with a little indirection"
+      (is (= {:projections '("_id" "name" "category_id" "latitude" "longitude" "price" "bob"),
+              :query
+              [{"$project"
+                {"_id" "$_id",
+                 "name" "$name",
+                 "category_id" "$category_id",
+                 "latitude" "$latitude",
+                 "longitude" "$longitude",
+                 "price" "$price",
+                 "bob" {"$abs" {"$subtract" ["$price" 300]}}}}
+               {"$limit" 5}],
+              :collection "venues",
+              :mbql? true}
+             (qp/query->native
+               (mt/mbql-query venues
+                              {:filters     [[:expression "bob"] [:expression "cobb"]]
+                               :expressions {:bob   [:abs [:- $price 300]]}
+                               :limit       5})))))
+
+    (testing "Should be able to deal with a little indirection, with an expression in"
+      (is (= {:projections '("_id" "name" "category_id" "latitude" "longitude" "price" "bob" "cobb"),
+              :query
+              [{"$project"
+                {"_id" "$_id",
+                 "name" "$name",
+                 "category_id" "$category_id",
+                 "bob" {"$abs" "$latitude"},
+                 "cobb" {"$ceil" {"$abs" "$latitude"}},
+                 "longitude" "$longitude",
+                 "latitude" "$latitude",
+                 "price" "$price"}}
+               {"$limit" 5}],
+              :collection "venues",
+              :mbql? true}
+             (qp/query->native
```

Comment:
I think these tests could look a little nicer if you had a nice helper function and it was called like
```clojure
(is (= {"bob"  {"$abs" "$latitude"},
        "cobb" {"$ceil" {"$abs" "$latitude"}}}
       (some-test-fn {:bob   [:abs (mt/id :latitude)]
                      :cobb  [:ceil [:expression "bob"]]})))
```

Reply:
not 100% a fan of terseness in testing but I guess making it terse will make it conform better to rest of file so sgtm